src/LegacyMongolidModel.php

```diff
@@ -0,0 +1,378 @@
+<?php
+namespace MongolidLaravel;
+
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Contracts\Validation\Factory as ValidationFactory;
+use Illuminate\Support\MessageBag;
+use Mockery;
+use Mockery\Expectation;
+use MongoDB\Collection;
+use Mongolid\Cursor\CursorInterface;
+use Mongolid\LegacyRecord;
+
+/**
+ * This class extends the Mongolid\LegacyRecord, so, in order
+ * to understand the ODM implementation make sure to check the
+ * base class.
+ *
+ * The MongolidLaravel\MongolidModel simply extends the original
+ * and framework agnostic model of MongoLid and implements some
+ * validation rules using Laravel validation components.
+ *
+ * Remember, this package is meant to be used with Laravel while
+ * the "leroy-merlin\mongolid" is meant to be used with other frameworks
+ * or even without any.
+ */
+abstract class LegacyMongolidModel extends LegacyRecord
+{
+    /**
+     * Validation rules.
+     *
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * Error message bag.
+     *
+     * @var MessageBag
+     */
+    protected $errors;
+
+    /**
+     * Public static mock.
+     *
+     * @var Mockery\Mock
+     */
+    public static $mock;
+
+    /**
+     * Public local mock.
+     *
+     * @var Mockery\Mock
+     */
+    public $localMock;
+
+    /**
+     * List of attribute names which should be hashed on save. For
+     * example: array('password');.
+     *
+     * @var array
+     */
+    protected $hashedAttributes = [];
+
+    /**
+     * Save the model to the database if it's valid. This method also
+     * checks for the presence of the localMock in order to call the save
+     * method into the existing Mock in order not to touch the database.
+     *
+     * @param bool $force force save even if the object is invalid
+     */
+    public function save(bool $force = false): bool
+    {
+        if ($this->localMockHasExpectationsFor('save')) {
+            return $this->getLocalMock()->save();
+        }
```

Comment:
```suggestion
        if ($this->localMockHasExpectationsFor('save')) {
            $this->getLocalMock()->save();

            return true;
        }
```
Mockery has not been configured to return a boolean when `save` is called, so it will return `null`, breaking code that expects to return a boolean.

Reply:
It is not the test that defines the return of the save method, but the `shouldReceiveSave` method.

Example of usage:
```php
$model->shouldReceiveSave();
$model->save();
```

Reply:
My mistake. I didn't see that `localMock` is returned.

Reply:
My mistake. I didn't see that `localMock` is returned.